### PR TITLE
Latest release/Archive pages now accept any platform, plus fixes

### PR DESCRIPTION
**Pull request by Joe-Brady**
Related to #35 

CURRENTLY ON STAGING: http://staging.adoptopenjdk.net/releases.html

Completed the second step towards full abstraction in the JS to accept any defined platform.

The 'Latest' section of releases.js will now accept all platform that are defined in the 'platforms' array at the top of global.js.

**Note:** new platforms are just added as blocks to the Latest page. These blocks appear in a long row. This long row flows onto new lines when it reaches the defined maximum width on the page. I will add a low-priority issue to possibly change this layout in the future, because while it works, 4/5/6+ platforms could perhaps be displayed more efficiently.

### CODE REVIEW

**Comment by gdams** [comment]
@Joe-Brady can you add the s390x and ppc64le icons ?

File: dist/js/app.js [hunk 1]
```diff
@@ -309,10 +326,10 @@ function populateNightly() {
 
           // firstly, check if the platform name is recognised...
           if(thisPlatform != false) {
-            var thisFileExtension = getFileExt(thisPlatform); // get an uppercase file extension associated with this platform
 
             // secondly, check if the file has the expected file extension for that platform...
             // (this filters out all non-binary attachments, e.g. SHA checksums - these contain the platform name, but are not binaries)
+            var thisFileExtension = getFileExt(thisPlatform); // get the file extension associated with this platform
             if(a.indexOf((thisFileExtension.toUpperCase())) >= 0) {
```
Comment (karianna) [inline dist/js/app.js]: I know you really, really want rename `a` :-)

File: dist/js/app.js [hunk 2]
```diff
@@ -484,54 +501,49 @@ function populateLatest() {
...
-      var linuxDlButton = document.getElementById("linux-dl-button");
-      var windowsDlButton = document.getElementById("windows-dl-button");
-      var macDlButton = document.getElementById("mac-dl-button");
-      var linuxPlatformBlock = document.getElementById("latest-linux");
-      var windowsPlatformBlock = document.getElementById("latest-windows");
-      var macPlatformBlock = document.getElementById("latest-mac");
-
+      // for each asset attached to this release, check if it's a valid binary, then add a download block for it...
       assetCounter2 = 0;
-      assetArray.forEach(function() {     // iterate through the binaries attached to this release
```
Comment (karianna) [inline dist/js/app.js]: Usually when you have loop counters like `assetCounter` and a `assetCounter2` it's a scoping or naming code smell....

File: dist/js/app.js [hunk 3]
```diff
@@ -484,54 +501,49 @@ function populateLatest() {
...
-
-        } else if(a.indexOf("MAC") >= 0) {
-          document.getElementById("latest-size-mac").innerHTML = Math.floor((assetArray[assetCounter2].size)/1024/1024);
-          document.getElementById("latest-checksum-mac").href = (assetArray[assetCounter2].browser_download_url).replace("tar.gz", "sha256.txt");
-
-          var macLink = (assetArray[assetCounter2].browser_download_url);
-          macDlButton.href = macLink;
-          macPlatformBlock.className = macPlatformBlock.className.replace( /(?:^|\s)hide(?!\S)/g , '' );
+        var a = nameOfFile.toUpperCase(); // make the name of the asset uppercase
+        var thisPlatform = getSearchableName(a); // get the searchableName, e.g. MAC or X64_LINUX.
```
Comment (karianna) [inline dist/js/app.js]: You know what I'm going to say here :-)

**Comment by Joe-Brady** [comment]
@gdams @karianna - Thanks for the reviews, I'll fix those bits as part of some other incoming changes to this PR! (Please don't merge this PR yet)

File: dist/js/app.js (hunk 2 above)
Comment (Joe-Brady) [inline dist/js/app.js]: Opened issue #65 for this

**Comment by Joe-Brady** [comment]
@gdams @karianna 
Ok, so I've done a lot of changes since the last time you checked:

For most of the changes, check out the names of each commit. Most notable is the (now fully populated) Archive page, with a brand-new, shiny platform filter drop-down.

Just to also note that the var 'a' is now named properly, and George's request for the logos is also complete.

**Pull request merged**